documentation/api/query/tutorial-pql.markdown

```diff
@@ -0,0 +1,403 @@
+
+    resources {
+      tag = "foo" and
+      exported = true
+    }
+
+This query returns resources whose set of tags *contains* the tag
+"foo", and which are exported.
+
+For easy reference, the full set of queryable attributes can be found in [the resource
+endpoint documentation][resources].
+
+### Regular expressions
+
+What if we want to restrict our results to a certain subset of nodes? We could use something like this:
+
+    resources {
+      certname = "www1.example.com" or
+      certname = "www2.example.com" or
+      certname = "www3.example.com"
+    }
+
+And this works great if we know exactly the set of nodes we want. But what if
+we want all the 'www' servers, regardless of how many we have? In this case, we
+can use the regular expression match operator `~`:
+
+    resources {
+      certname ~ 'www\d+.example.com'
+    }
+
+### Choosing fields to return
+
+When you execute a resources query, all available fields for that query are returned
+by default. Using the projection syntax however, you can limit what fields are returned.
+
+For example, if you want to only respond with the `certname`, `type`, and `title` of
+each resource where Class[apache] is defined, you can do the folllowing:
+
+    resources[certname, type, title] {
+      type = "Class" and
+      title = "apache"
+    }
+
+## Facts walkthrough
+
+In addition to resources, we can also query for facts. This looks similar,
+though the available fields and operators are a bit different. Some things are
+the same, though. For instance, suppose you want all the facts for a certain
+node:
+
+    facts { certname = "foo.example.com" }
+
+This gives results that look something like this:
+
+    [ {
+      "certname" : "foo.example.com",
+      "name" : "architecture",
+      "value" : "amd64",
+      "environment" : "production"
+    }, {
+      "certname" : "foo.example.com",
+      "name" : "fqdn",
+      "value" : "foo.example.com",
+      "environment" : "production"
+    }, {
+      "certname" : "foo.example.com",
+      "name" : "hostname",
+      "value" : "foo",
+      "environment" : "production"
+    }, {
+      "certname" : "foo.example.com",
+      "name" : "ipaddress",
+      "value" : "192.168.100.102",
+      "environment" : "production"
+    }, {
+      "certname" : "foo.example.com",
+      "name" : "kernel",
+      "value" : "Linux",
+      "environment" : "production"
+    }, {
+      "certname" : "foo.example.com",
+      "name" : "kernelversion",
+      "value" : "2.6.32",
+      "environment" : "production"
+    } ]
+
+### Fact attributes
+
+In the last query, we saw that a "fact" consists of a "certname", a "name", and
+a "value". As you might expect, we can query using "name" or "value".
+
+    facts {
+      name = "operatingsystem" and
+      value = "Debian"
+    }
+
+This will find all the "operatingsystem = Debian" facts, and their
+corresponding nodes. As you see, "and" is supported for facts, as are "or" and
+"not".
```

Comment:
pretty minor but we've got quotes on operators here and backticks on the next line